src/utils/dynamo.py

```diff
@@ -22,11 +21,28 @@ def write_to_dynamo(
 
     #######################################################################################################################################
     # USED FOR TESTING IN LAMBDA. Include 'testSuffix' in your lambda test object so that the dynamo items used in prod are not overwritten
+    isTesting = False
+    testSuffix = ""
+
     if event != "":
-        testSuffix = "" if ("testSuffix" not in event) else event["testSuffix"]
-        country += testSuffix
+        if ("testSuffix" in event) and (event["testSuffix"] != ""):
+            testSuffix = event["testSuffix"]
+            isTesting = True
     #######################################################################################################################################
 
+    # Get the existing object, that object will be PUT back into dynamo, but marked as 'old' so it can be compared.
+    # The comparison will be used to determine if the translator needs to translate any of the newly scraped data or not.
+    existingItem = get_existing_object(country)
+
+    # If we're testing we're find to GET and object out of dynamo
+    # But we don't want to overwrite the 'old' version of the object that is used for comparison
+    # (which is used to tell us if we need to translate it or not)
+    if isTesting:
+        existingItem["country"]["S"] = existingItem["country"]["S"] + testSuffix
+
+    # Mark the existing item as 'old', then scrape to get the most 'up to date' information.
+    update_existing_item_as_old(existingItem)
```

Comment:
If the code fails at a later part in this function, how do we perform a rollback?

Reply:
As far as I can see, there's not much to fail (unless inserting the new values into the db fails, which would mean rolling back wouldn't be possible anyway). Also I'm pretty sure it just makes a copy of the current db entry, so all the data should still be present after the copy.

Reply:
Performing rollback would be as easy as removing '-old' from the object in dynamo. The real problem is, how do we know it failed? We'll have to figure that out sometime soon probably

Reply:
I think down the line we can explore using something like [PyDynamoDB](https://github.com/pynamodb/PynamoDB) for an ORM in the future, I'm ok to merge first